fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -1739,14 +1729,19 @@ private ExtractedSearchParameters extractSearchParameters(Resource fhirResource,
                 // Augment the extracted parameter list with special values we use to represent compartment relationships.
                 // These references are stored as tokens and are used by the search query builder
                 // for compartment-based searches
-                addCompartmentParams(allParameters, fhirResource);
+                addCompartmentParams(allParameters, fhirResource.getClass().getSimpleName());
 
                 // If this is a definitional resource, augment the extracted parameter list with a composite
                 // parameter that will be used for canonical searches. It will contain the url and version
                 // values.
                 addCanonicalCompositeParam(allParameters);
             }
 
+            // Remove parameters that aren't to be stored
+            boolean anyRemoved = allParameters.removeIf(value -> !value.isForStoring());
```

Comment:
removeIf isn't all that cheap on an ArrayList. Might be worth considering a linked list for allParameters - which has much lower remove costs.

Reply:
i made the change